src/container/RelayRenderer.js

```diff
@@ -342,12 +287,50 @@ class RelayRenderer extends React.Component {
     this.mounted = false;
   }
 
+  /**
+   * @private
+   */
+  _resolveContainerProps(): Object {
+    return this.containerProps = (
+      this.containerProps ||
+      {
+        ...this.props.queryConfig.params,
+        ...mapObject(
+          this.querySet,
+          query => createFragmentPointerForRoot(this.props.environment, query)
+        ),
+      }
+    );
+  }
+
+  /**
+   * @private
+   */
+  _calculateRenderArgs(): RelayRendererRenderArgs {
+    const {readyState} = this.state;
+    return readyState ?
+      {
+        done: readyState.done,
+        error: readyState.error,
+        props: readyState.ready ? this._resolveContainerProps() : null,
+        retry: this._retry.bind(this),
+        stale: readyState.stale,
+      } :
+      {
+        done: false,
+        error: null,
+        props: null,
+        retry: this._retry.bind(this),
+        stale: false,
+      };
```

Comment:
But isn't it good practice to follow functional programming style when possible? Besides there is even fancier ternary in `_runQueries`. :smile: